prov/rxm/src/rxm_conn.c

```diff
@@ -661,10 +654,7 @@ static int rxm_cmap_cm_thread_close(struct rxm_cmap *cmap)
 			"Unable to signal CM thread\n");
 		return ret;
 	}
-	/* Release lock so that CM thread could process shutdown events */
-	cmap->release(&cmap->lock);
 	ret = pthread_join(cmap->cm_thread, NULL);
```

Comment:
Based on the code changes, I'm assuming that we're holding the cmap->lock here.  If the cm_thread tries to acquire that same lock, it will block, and pthread_join will hang.